### PR TITLE
Fix progress metric registration and parsing

### PR DESCRIPTION
**Pull request by arnongilboa**
**What this PR does / why we need it**:
Use default metric registration. We shouldn't use the controller-runtime registration as we have no controller here and it will not register the metric correctly.

Add `kubevirt_cdi_import_progress_total` metric and use it in the importer instead of `kubevirt_cdi_clone_progress_total`.

Fix the metric parsing to match their new names. Otherwise the DV progress will not be updated until its 100%.

Regression introduced in #3254

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
Fix progress metric registration and parsing
```

### CODE REVIEW

**Comment by arnongilboa** [comment]
/hold

Should fix utest and improve func test.

File: pkg/monitoring/metrics/cdi-cloner/metrics.go [hunk 1]
```diff
@@ -2,14 +2,10 @@ package cdicloner
...
 import (
 	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
-
-	runtimemetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
 )
 
 // SetupMetrics register prometheus metrics
 func SetupMetrics() error {
-	operatormetrics.Register = runtimemetrics.Registry.Register
-	operatormetrics.Unregister = runtimemetrics.Registry.Unregister
```
Comment (akalenyu) [inline pkg/monitoring/metrics/cdi-cloner/metrics.go:12]: Isn't the registerer in `operatormetrics` the same as the one in ctrl runtime? just trying to understand what really breaks here

**Comment by akalenyu** [comment]
Note there are a couple more of these regex spread around. Do we want to maybe unify them?
```bash
$ (cd pkg; git grep 'regexp.MustCompile("progress'\\\\)
controller/common/util.go:      importRegExp := regexp.MustCompile("progress\\{ownerUID\\=\"" + args.OwnerUID + "\"\\} (\\d{1,3}\\.?\\d*)")
controller/populators/import-populator.go:      importRegExp := regexp.MustCompile("progress\\{ownerUID\\=\"" + string(pvc.UID) + "\"\\} (\\d{1,3}\\.?\\d*)")
```

File: pkg/controller/datavolume/controller-base.go [hunk 2]
```diff
@@ -1109,7 +1109,7 @@ func updateProgressUsingPod(dataVolumeCopy *cdiv1.DataVolume, pod *corev1.Pod) e
 	}
 
 	// Example value: import_progress{ownerUID="b856691e-1038-11e9-a5ab-525500d15501"} 13.45
-	var importRegExp = regexp.MustCompile("progress\\{ownerUID\\=\"" + string(dataVolumeCopy.UID) + "\"\\} (\\d{1,3}\\.?\\d*)")
+	var importRegExp = regexp.MustCompile("progress_total\\{ownerUID\\=\"" + string(dataVolumeCopy.UID) + "\"\\} (\\d{1,3}\\.?\\d*)")
```
[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/controller/datavolume/controller-base.go:1112]: The import progress always used `clone_progress`, so I did not split it, although we may consider renaming it to `kubevirt_cdi_progress_total` dropping the `clone`?

**Comment by arnongilboa** [comment]
> Note there are a couple more of these regex spread around. Do we want to maybe unify them?
> 
> ```shell
> $ (cd pkg; git grep 'regexp.MustCompile("progress'\\\\)
> controller/common/util.go:      importRegExp := regexp.MustCompile("progress\\{ownerUID\\=\"" + args.OwnerUID + "\"\\} (\\d{1,3}\\.?\\d*)")
> controller/populators/import-populator.go:      importRegExp := regexp.MustCompile("progress\\{ownerUID\\=\"" + string(pvc.UID) + "\"\\} (\\d{1,3}\\.?\\d*)")
> ```

Good catch, I'll sure unify them.

File: pkg/monitoring/metrics/cdi-cloner/metrics.go (hunk 1 above)
Comment (arnongilboa) [reply pkg/monitoring/metrics/cdi-cloner/metrics.go:12]: @machadovilaca can you please clarify the difference between the registerers?

Comment (machadovilaca) [reply pkg/monitoring/metrics/cdi-cloner/metrics.go:12]: By default the `operatormetrics` uses the same registry as the Prometheus client. `controller-runtime` creates it's own custom registry, that's why controllers using the `controller-runtime` metrics endpoint need to do `operatormetrics.Register = runtimemetrics.Registry.Register`

**Comment by akalenyu** [comment]
Can I rebase https://github.com/kubevirt/containerized-data-importer/pull/3293 on this?

**Comment by arnongilboa** [comment]
> Can I rebase #3293 on this?
I'm doing a small API refactoring, so wait a few mins.

File: pkg/controller/datavolume/controller-base.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/controller/datavolume/controller-base.go:1113]: Sure, I'll fix it.

[the remark this answers is not in the document]
Comment (akalenyu) [reply pkg/controller/datavolume/controller-base.go:1113]: Maybe it makes sense to have a func to unify all the similar regex?

[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/controller/datavolume/controller-base.go:1113]: Sure, that's what I am doing.

**Review by akalenyu — COMMENTED** [review]
Looks good!

File: pkg/controller/common/util.go [hunk 3]
```diff
@@ -328,6 +328,15 @@ const (
...
 	LabelDynamicCredentialSupport = "kubevirt.io/dynamic-credentials-support"
 
+	// ImportProgressMetricName is the name of the import progress metric
+	ImportProgressMetricName = "kubevirt_cdi_import_progress_total"
+	// CloneProgressMetricName is the name of the clone progress metric
+	CloneProgressMetricName = "kubevirt_cdi_clone_progress_total"
+	// OpenStackPopulatorProgressMetricName is the name of the OpenStack populator progress metric
+	OpenStackPopulatorProgressMetricName = "kubevirt_cdi_openstack_populator_progress_total"
+	// OvirtPopulatorProgressMetricName is the name of the oVirt populator progress metric
+	OvirtPopulatorProgressMetricName = "kubevirt_cdi_ovirt_progress_total"
```
[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/controller/common/util.go:338]: Right. I was just hesitating about the cross package ref just for the names.

File: pkg/controller/populators/forklift-populator.go [hunk 4]
```diff
@@ -423,9 +422,9 @@ func (r *ForkliftPopulatorReconciler) updateImportProgress(podPhase string, pvc,
 	}
 
 	// We fetch the import progress from the import pod metrics
-	importRegExp := regexp.MustCompile("progress_total\\{ownerUID\\=\"" + string(pvc.UID) + "\"\\} (\\d{1,3}\\.?\\d*)")
 	httpClient = cc.BuildHTTPClient(httpClient)
-	progressReport, err := cc.GetProgressReportFromURL(url, importRegExp, httpClient)
+	progressReport, err := cc.GetProgressReportFromURL(url, httpClient,
+		cc.OpenStackPopulatorProgressMetricName+"|"+cc.OvirtPopulatorProgressMetricName, string(pvc.UID))
```
Comment (akalenyu) [inline pkg/controller/populators/forklift-populator.go:427]: printf will look nicer no?

File: pkg/controller/common/util.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (akalenyu) [reply pkg/controller/common/util.go:338]: :scream: is there a circular dep?

[the remark this answers is not in the document]
Comment (arnongilboa) [reply pkg/controller/common/util.go]: no no...

**Pull request merged**